execution/executor/tests/db_bootstrapper_test.rs

```diff
@@ -244,3 +271,78 @@ fn test_pre_genesis() {
     // Pre-genesis state accessible.
     assert_eq!(get_balance(&account2, &db_rw), 2000);
 }
+
+#[test]
+fn test_new_genesis() {
+    let (config, genesis_key) = config_builder::test_config();
+
+    // Create bootstrapped DB.
+    let tmp_dir = TempPath::new();
+    let db = DbReaderWriter::new(LibraDB::new(&tmp_dir));
+    let genesis_txn = get_genesis_txn(&config).unwrap();
+    bootstrap_db_if_empty::<LibraVM>(&db, genesis_txn).unwrap();
+
+    // Mint for 2 demo accounts.
+    let (account1, account1_key, account2, account2_key) = get_demo_accounts();
+    let txn1 = get_mint_transaction(&genesis_key, 1, &account1, &account1_key, 2_000_000);
+    let txn2 = get_mint_transaction(&genesis_key, 2, &account2, &account2_key, 2_000_000);
+    execute_and_commit(vec![txn1, txn2], &db);
+    assert_eq!(get_balance(&account1, &db), 2_000_000);
+    assert_eq!(get_balance(&account2, &db), 2_000_000);
+    let (li, validator_change_proof, _) = db.reader.get_state_proof(0).unwrap();
+    let trusted_state = TrustedState::new_trust_any_genesis_WARNING_UNSAFE();
+    trusted_state
+        .verify_and_ratchet(&li, &validator_change_proof)
+        .unwrap();
+
+    // New genesis transaction: set validator set and overwrite account1 balance.
+    let genesis_txn = Transaction::WaypointWriteSet(ChangeSet::new(
+        WriteSetMut::new(vec![
+            (
+                ValidatorSet::CONFIG_ID.access_path(),
+                WriteOp::Value(lcs::to_bytes(&ValidatorSet::new(vec![])).unwrap()),
+            ),
+            (
+                AccessPath::new(account1, BalanceResource::resource_path()),
+                WriteOp::Value(lcs::to_bytes(&BalanceResource::new(1_000_000)).unwrap()),
+            ),
+        ])
+        .freeze()
+        .unwrap(),
+        vec![ContractEvent::new(
+            on_chain_config::new_epoch_event_key(),
+            0,
+            lbr_type_tag(),
+            vec![],
+        )],
+    ));
+
+    // Bootstrap DB into new genesis.
+    let tree_state = db.reader.get_latest_tree_state().unwrap();
+    let waypoint = bootstrap_db::<LibraVM>(&db, tree_state, &genesis_txn).unwrap();
```

Comment:
what's the difference between this test and the test_pre_genesis? I saw both are overwriting existing state?

Reply:
the other one boots from empty accumulator, this from existing accumulator and ledger info.